activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-api/src/main/java/org/activiti/cloud/modeling/api/ModelValidationErrorProducer.java

```diff
@@ -72,4 +72,20 @@ default ModelValidationError createModelValidationError(boolean warning,
         modelValidationError.setErrorCode(errorCode);
         return modelValidationError;
     }
+
+    default ModelValidationError createModelValidationError(boolean warning,
```

Comment:
Stupid question (and maybe out of the scope of this PR), but why do we have this kind of method here in this interface? Wouldn't it be better to provide directly different constructors in `ModelValidationError`? So instead of calling `createModelValidationError(problem, description, schema)` we simply call `new ModelValidationError(problem, description, schema)`.

In any case, if you are adding a new method here it should reuse the previous one and only set the additional `referenceId`.

```java
default ModelValidationError createModelValidationError(boolean warning,
															String problem,
                                                            String description,
                                                            String schema,
                                                            String referenceId,
                                                            String errorCode) {
        ModelValidationError modelValidationError = createModelValidationError(warning, problem, description, schema, errorCode);
        modelValidationError.setReferenceId(referenceId);
        return modelValidationError;
    }

Reply:
My apologies, I forgot to convert the PR into draft as am still making changes.  I will make it ready once am done.  Thanks.

Reply:
Oops, I reviewed it too early then :)